### PR TITLE
Support for MaxMessageSizeInKilobytes

### PR DESCRIPTION
**Pull request by ki1729**
- Adding support for max message size feature
- Move to service API version 2021-05

### CODE REVIEW

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/administration/ServiceBusAdministrationAsyncClient.java
Comment (ki1729) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/administration/ServiceBusAdministrationAsyncClient.java:9]: Please ignore this file - got included due to bad git branch structuring - will be fixed as soon as the 25146 PR is merged, this will become clean

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/administration/ServiceBusAdministrationClientBuilder.java
Comment (ki1729) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/administration/ServiceBusAdministrationClientBuilder.java:127]: Please ignore this file - got included due to bad git branch structuring - will be fixed as soon as the 25146 PR is merged, this will become clean

File: sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/administration/ServiceBusAdministrationAsyncClientTest.java
Comment (ki1729) [inline sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/administration/ServiceBusAdministrationAsyncClientTest.java:6]: Other than the createQueue Test, the remaining changes are for 25146 PR, please ignore the rest of the changes

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/administration/ServiceBusAdministrationClientBuilder.java
Comment (joshfree) [reply sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/administration/ServiceBusAdministrationClientBuilder.java:127]: should we hold off on this PR until the other one is merged and this is rebased?

Comment (joshfree) [reply sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/administration/ServiceBusAdministrationClientBuilder.java:127]: adding `Do Not Merge` label to denote that there's some git cleanup necessary.  Please remove the label when the PR is ready :)

File: sdk/servicebus/azure-messaging-servicebus/CHANGELOG.md
Comment (joshfree) [inline sdk/servicebus/azure-messaging-servicebus/CHANGELOG.md:7]: nit: could this reference `quoted` concrete types similar to past changelog entries?

**Review by joshfree — CHANGES_REQUESTED** [review]
Holding off on reviewing rest of file until PR is rebased to exclude other unrelated changes

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusServiceVersion.java
Comment (ramya-rao-a) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusServiceVersion.java:11]: When the customer targets this older API version, the maxMessageSizeInKilobytes property will be null in the response from the service. JS & .NET had to change this property to be nullable to support this case. Would you need a similar approach in Java?

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/administration/models/CreateQueueOptions.java
Comment (conniey) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/administration/models/CreateQueueOptions.java:481]: nit: remove en-us so user's browser can discern which language to use.

File: sdk/servicebus/azure-messaging-servicebus/CHANGELOG.md
Comment (ki1729) [reply sdk/servicebus/azure-messaging-servicebus/CHANGELOG.md:7]: done

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/administration/models/CreateQueueOptions.java
Comment (ki1729) [reply sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/administration/models/CreateQueueOptions.java:481]: nice! didn't know this neat trick

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusServiceVersion.java
Comment (ki1729) [reply sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusServiceVersion.java:11]: this is not required property in Java, so we should be good here

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/administration/models/CreateQueueOptions.java
Comment (joshfree) [reply sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/administration/models/CreateQueueOptions.java]: it should actually fail CI if we have any embedded URIs to language-specific docs sites

File: sdk/servicebus/azure-messaging-servicebus/CHANGELOG.md
Comment (joshfree) [inline sdk/servicebus/azure-messaging-servicebus/CHANGELOG.md:6]: nit: would be nice if everything was in past tense for consistency (moved)

Comment (ki1729) [reply sdk/servicebus/azure-messaging-servicebus/CHANGELOG.md:6]: ack

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/administration/models/CreateTopicOptions.java
Comment (joshfree) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/administration/models/CreateTopicOptions.java:422]: remove en-us

File: sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/implementation/ServiceBusManagementSerializerTest.java
Comment (joshfree) [inline sdk/servicebus/azure-messaging-servicebus/src/test/java/com/azure/messaging/servicebus/implementation/ServiceBusManagementSerializerTest.java:479]: @conniey should these services be moved to a team-wide resource (not something with coniey)?

**Review by joshfree — APPROVED** [review]
reviewed; please get a review from @anuchandy and/or @conniey before merging; Thanks

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/administration/models/CreateQueueOptions.java
Comment (JonathanGiles) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/administration/models/CreateQueueOptions.java:487]: Do we worry about the different 'in*' methods in this class (inKilobytes and inMegabytes)? (It seems unlikely to be a concern, but I just wanted to double-check)

**Review by JonathanGiles — APPROVED** [review]

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/administration/models/CreateQueueOptions.java
Comment (anuchandy) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/administration/models/CreateQueueOptions.java:485]: nit; the return doc

> the CreateQueueOptions object itself.

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/administration/models/CreateTopicOptions.java
Comment (anuchandy) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/administration/models/CreateTopicOptions.java:427]: nit; the return doc

> the CreateTopicOptions object itself.

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/administration/models/QueueProperties.java
Comment (anuchandy) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/administration/models/QueueProperties.java:137]: Above Ramya mentioned the service may return `null` for this property when targetting older API versions. If that's the case, then we're converting that `null` to`0L` here, if so, when we send the same model payload to service for an update, we will be sending `0L`, which might cause the service to throw an error.

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/administration/models/CreateTopicOptions.java
Comment (ki1729) [reply sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/administration/models/CreateTopicOptions.java:422]: done

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/administration/models/CreateQueueOptions.java
Comment (ki1729) [reply sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/administration/models/CreateQueueOptions.java:487]: Thanks Jonanthan! Maybe in the future, currently message sizes are capped to 100MB on the premium tier, so we should okay here keeping to the convention of using kB

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/administration/models/CreateTopicOptions.java
Comment (ki1729) [reply sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/administration/models/CreateTopicOptions.java:427]: done

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/administration/models/CreateQueueOptions.java
Comment (ki1729) [reply sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/administration/models/CreateQueueOptions.java:485]: done

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/administration/models/QueueProperties.java
Comment (ki1729) [reply sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/administration/models/QueueProperties.java:139]: Yeah Anu, as long as we don't send the parameter when it is 0L we should be good here. And that is what we do, before doing a create / update we only include this parameter if it is non-zero

**Review by anuchandy — APPROVED** [review]

**Pull request merged**
